### PR TITLE
Only refresh primary key when a sequence is attached

### PR DESCRIPTION
**Pull request by shayonj**
This is for when a column is smallint, bigint, int. Then, we only do a +1 on the PK during swap for those column types. For other column types, there won't be any sequence, so we can rely on this check accordingly

Follow on from: https://github.com/shayonj/pg-osc/pull/72

Closes: https://github.com/shayonj/pg-osc/issues/71

### CODE REVIEW

**Comment by shayonj** [comment]
@jfrost

File: lib/pg_online_schema_change/query.rb [hunk 1]
```diff
@@ -319,6 +319,28 @@ def copy_data_statement(client, shadow_table, reuse_trasaction = false)
...
+        end&.first
+      end
+
+      def query_for_primary_key_refresh(shadow_table, primary_key, table, opened)
+        sequence_name = primary_key_sequence(shadow_table, primary_key, opened)
+
+        return "" if sequence_name.nil?
+
+        <<~SQL
+          SELECT setval((select pg_get_serial_sequence(\'#{shadow_table}\', \'#{primary_key}\')), (SELECT max(#{primary_key}) FROM #{table})+1);
```
Comment (brycethornton) [inline lib/pg_online_schema_change/query.rb:341]: In my testing it appears that the `+1` causes a skipped sequence value. I think `setval()` sets the current value (which we get via the `max()` call) which will allow `nextval()` to return the proper next value in the sequence.

https://www.postgresql.org/docs/current/functions-sequence.html

Comment (shayonj) [reply lib/pg_online_schema_change/query.rb:341]: ah! nice catch @brycethornton . I have a failing spec now: https://github.com/shayonj/pg-osc/pull/74/files

Should have a fix soon.

Comment (shayonj) [reply lib/pg_online_schema_change/query.rb:341]: @jfrost pointed this out here as well: https://github.com/shayonj/pg-osc/pull/72#discussion_r973605106

**Pull request merged**